tests/components/camera/test_webrtc.py

```diff
@@ -231,6 +280,690 @@ async def test_ws_get_client_config_no_rtc_camera(
+        assert response["id"] == subscription_id
+        assert response["type"] == "event"
+        assert response["event"]["type"] == "session_id"
+        session_id = response["event"]["sessionId"]
+
+        send_message(message)
+
+        response = await client.receive_json()
+        assert response["id"] == subscription_id
+        assert response["type"] == "event"
+        assert response["event"] == expected_frontend_message
+
+        # Unsubscribe/Close session
+        await client.send_json_auto_id(
+            {
+                "type": "unsubscribe_events",
+                "subscription": subscription_id,
+            }
+        )
+        msg = await client.receive_json()
+        assert msg["success"]
+        mock_async_close_session.assert_called_once_with(session_id)
+
+
+@pytest.mark.usefixtures("mock_camera_webrtc")
+async def test_websocket_webrtc_offer_invalid_entity(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test WebRTC with a camera entity that does not exist."""
+    client = await hass_ws_client(hass)
+    await client.send_json_auto_id(
+        {
+            "type": "camera/webrtc/offer",
+            "entity_id": "camera.does_not_exist",
+            "offer": WEBRTC_OFFER,
+        }
+    )
+    response = await client.receive_json()
+
+    assert response["type"] == TYPE_RESULT
+    assert not response["success"]
+    assert response["error"] == {
+        "code": "home_assistant_error",
+        "message": "Camera not found",
+    }
+
+
+@pytest.mark.usefixtures("mock_camera_webrtc")
+async def test_websocket_webrtc_offer_missing_offer(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test WebRTC stream with missing required fields."""
+    client = await hass_ws_client(hass)
+    await client.send_json_auto_id(
+        {
+            "type": "camera/webrtc/offer",
+            "entity_id": "camera.demo_camera",
+        }
+    )
+    response = await client.receive_json()
+
+    assert response["type"] == TYPE_RESULT
+    assert not response["success"]
+    assert response["error"]["code"] == "invalid_format"
+
+
+@pytest.mark.parametrize(
+    ("error", "expected_message"),
+    [
+        (ValueError("value error"), "value error"),
+        (HomeAssistantError("offer failed"), "offer failed"),
+        (TimeoutError(), "Timeout handling WebRTC offer"),
+    ],
+)
+@pytest.mark.usefixtures("mock_camera_webrtc_frontendtype_only")
+async def test_websocket_webrtc_offer_failure(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    error: Exception,
+    expected_message: str,
+) -> None:
+    """Test WebRTC stream that fails handling the offer."""
+    client = await hass_ws_client(hass)
+
+    camera_obj = get_camera_from_entity_id(hass, "camera.demo_camera")
+    camera_obj._webrtc_sync_offer = True  # Setting it to True to simulate async_handle_web_rtc_offer would be overwritten
```

Comment:
The way i've seen recommended to handle this is to stop using the demo platform and instead use a real entity created in the test but with a mock platform.